### PR TITLE
fix: docbuild unexpected section titles

### PR DESCRIPTION
**Pull request by ethanglaser**
## Description

_Add a comprehensive description of proposed changes_

_List associated issue number(s) if exist(s): #6 (for example)_

_Documentation PR (if needed): #1340 (for example)_

_Benchmarks PR (if needed): https://github.com/IntelPython/scikit-learn_bench/pull/155 (for example)_

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] I have run it locally and tested the changes extensively.
- [ ] All CI jobs are green or I have provided justification why they aren't.
- [ ] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.
- [ ] I have provided justification why quality metrics have changed or why changes are not expected.
- [ ] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

**Comment by icfaust** [comment]
@ethanglaser thoroughly impressed, how did you find these?

File: sklearnex/basic_statistics/basic_statistics.py [hunk 1]
```diff
@@ -53,26 +53,26 @@ class BasicStatistics(oneDALEstimator, BaseEstimator):
 
     Attributes
     ----------
-        min_ : ndarray of shape (n_features,)
```
Comment (david-cortes-intel) [inline sklearnex/basic_statistics/basic_statistics.py:56]: This is a valid format and is rendered correctly in all the docs I've seen. Were you seeing some issue with these somewhere?

File: sklearnex/_utils.py [hunk 2]
```diff
@@ -202,8 +202,8 @@ def _add_inc_serialization_note(class_docstrings: str) -> str:
     # In python versions >=3.13, leading whitespace in docstrings defined through
     # static strings (but **not through other ways**) is automatically removed
     # from the final docstrings, while in earlier versions is kept.
-    inc_serialization_note = """Note
-----
+    inc_serialization_note = """Notes
```
Comment (david-cortes-intel) [inline sklearnex/_utils.py:205]: 'Note' is a recognized section name (can be verified that it renders correctly in the artifacts from docs CI), and there can be multiple ones within a doc, whether sequential or interleaved with other titles. Did this produce an error somewhere?

Comment (david-cortes-intel) [reply sklearnex/_utils.py:205]: For reference:
image

File: sklearnex/basic_statistics/basic_statistics.py (hunk 1 above)
Comment (david-cortes-intel) [reply sklearnex/basic_statistics/basic_statistics.py:56]: For reference:
image

**Comment by ethanglaser** [comment]
> @ethanglaser thoroughly impressed, how did you find these?

Don't be too impressed yet because its still failing :(

I got really into the weeds trying to figure out why this was happening yesterday - I am 90% sure its not related to a PR merged on our end and there is no definitive error message in the docbuild logs. So I tried addressing these unexpected section title messages in the logs since they appear to be new, and found that scikit looks for very specific formatting in those and we have a couple minor discrepancies. But they are still failing...

File: sklearnex/_utils.py (hunk 2 above)
Comment (ethanglaser) [reply sklearnex/_utils.py:205]: Yeah seems like this update did not improve the status so it will likely be dropped

**Comment by ethanglaser** [comment]
I guess you guys have probably noticed but all Docs jobs have failed consistently since Friday: https://dev.azure.com/daal/daal4py/_build?definitionId=19

There are messages like this in the log that seem to be the issue:

/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/sklearnex/covariance/incremental_covariance.py:docstring of sklearn.utils._metadata_requests.RequestMethod.__get__.<locals>.func:31: CRITICAL: Unexpected section title.

Returns
------- [docutils]

Observations are that it only complains on estimators that are not present in stock scikit-learn (BasicStatistics, incremental non-PCA ones) and that the message occurs n-1 times per estimator where n is the number of times that `Returns` and `Parameters` are present in the file

**Comment by icfaust** [comment]
@ethanglaser @david-cortes-intel Comes from sklearn 1.7.1 release: scikit-learn/scikit-learn#31804

EDIT: Looks like scikit-learn/scikit-learn#31805 solves the issue, but won't come available to us until 1.8, I'd recommend hardcoding sklearn to 1.7.0 until the 1.8 release, with some sort of poison pill test to change it when 1.8 drops.

**Comment by ethanglaser** [comment]
> @ethanglaser @david-cortes-intel Comes from sklearn 1.7.1 release: [scikit-learn/scikit-learn#31804](https://github.com/scikit-learn/scikit-learn/issues/31804)

Nice! Weird that I started seeing it in PRs merged before our 1.7.1 update (ie https://github.com/uxlfoundation/scikit-learn-intelex/runs/46250153550)

**Comment by icfaust** [comment]
@ethanglaser it was purposefully set to unlocked (https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/requirements-doc.txt#L52) by @david-cortes-intel in #2603 which means its not getting tracked by renovatebot (I think)

**Comment by ethanglaser** [comment]
Okay that explains a lot...
